Shooting/Shooting/Scene/GameScene/Player/Object/Missile.cpp

```diff
@@ -0,0 +1,108 @@
+﻿#include "Missile.h"
+#include "GameLib/GameLib.h"
+
+namespace {
+	const float MissileWidth = 16.0f;
+	const float MissileHeight = 32.0f;
+	const float MissileDeletePos_y = -10.0f;
+	const float MissileSpeed = 10.0f;
+}
+
+Missile::Missile()
+{
+	Initialize();
+}
+
+Missile::~Missile()
+{
+	Release();
+}
+
+//初期化する
+bool Missile::Initialize()
+{
+	const char* TexturePath = "../Graphics/Missile.jpg";
+
+	if (!m_texture.Load(TexturePath))
+	{
+		return false;
+	}
+
+	return true;
+}
+
+//解放する
+void Missile::Release()
+{
+	m_texture.Release();
+	m_missiles.clear();
+	m_missiles.shrink_to_fit();
+}
+
+void Missile::Create(float missile_x, float missile_y)
+{
+	MissileInfo Missile(missile_x, missile_y);
+	m_missiles.push_back(Missile);
+}
+
+void Missile::Update()
+{
+	/*for (int i = 0; i < m_missiles.size(); i++)
+	{
+		m_missiles[i].y -= MissileSpeed;
+
+		HELPER_2D->SetVerticesFromLeftTopType(
+			m_missiles[i].vertex,
+			m_missiles[i].x,
+			m_missiles[i].y,
+			MissileWidth,
+			MissileHeight
+		);
+
+		if (m_missiles[0].y < MissileDeletePos_y)
+		{
+			m_missiles.erase(m_missiles.begin());
+		}	
+	}*/
+
+	for (auto& missile : m_missiles)
+	{
+		missile.y -= MissileSpeed;
+
+		HELPER_2D->SetVerticesFromLeftTopType(
+			missile.vertex,
+			missile.x,
+			missile.y,
+			MissileWidth,
+			MissileHeight
+		);
+
+		if (m_missiles[0].y < MissileDeletePos_y)
+		{
+			m_missiles.erase(m_missiles.begin());
+		}
+	}
+}
+
+void Missile::Render()
+{
+	IDirect3DDevice9* pDevice = GameLib::Instance.GetDirect3DDevice();
+
+	//頂点に入れるデータを設定
+	pDevice->SetFVF(FVF_SIMPLE_TEX_2D);
+
+	//テクスチャの設定
+	pDevice->SetTexture(0, m_texture.GetD3DTexture());
+
+	//描画
+	for (auto& missile : m_missiles)
+	{	  
+	    pDevice->DrawPrimitiveUP(D3DPT_TRIANGLEFAN, 2, missile.vertex, sizeof(Simple2DVertex));
+	}
+}
+
+Missile::MissileInfo::MissileInfo(float missile_x, float missile_y)
+{
+	MissileInfo::x = missile_x;
```

Comment:
```suggestion
	x = missile_x;
```
ではダメなの？

@yttmwork 
というか、ここで `MissileInfo::x` ってどういう意味になるんですか…？

Reply:
@taishi-matsumura  
お疲れ様です。  
サンプル作って見たところ複数インスタンスを生成しても
きちんと各インスタンス毎に値が保存されるので、
this->と同等な意味合いになるんですかね？

Reply:
@yttmwork 
おぉ、検証ありがとうございます。

だとしたら

Reply:
@yttmwork 
おぉ、検証ありがとうございます。

@rikita-fujizono 
だとしたら、 `this->x = x;` の方がわかりやすいですね。
`MissileInfo::x` と書かれると、僕みたいなオジさんは「staticなメンバにアクセスしてる」みたいな印象を持ってしまいます。

Reply:
変数などの名前の変更のコミットで直しました。